transport/http/server_test.go

```diff
@@ -393,6 +393,33 @@ func TestEnhancedError(t *testing.T) {
 	}
 }
 
+func TestNoOpRequestDecoder(t *testing.T) {
+
+	done := make(chan struct{})
+
+	handler := httptransport.NewServer(
+		func(ctx context.Context, request interface{}) (interface{}, error) {
+			if request != nil {
+				t.Error("Expected nil request in endpoint when using NoOpRequestDecoder")
+			}
+			close(done)
+			return nil, nil
+		},
+		httptransport.NoOpRequestDecoder,
+		httptransport.EncodeJSONResponse,
+	)
+
+	server := httptest.NewServer(handler)
+	defer server.Close()
+	go http.Get(server.URL)
+
+	select {
+	case <-done:
+	case <-time.After(time.Second):
+		t.Fatal("timeout waiting for finalizer")
+	}
```

Comment:
I think this test is doing too much. It seems to me that it would be perfectly valid to test this without spinning up an http server and extra goroutines. Consider calling `handler.ServeHTTP` directly. You can use `httptest.NewRequest` and `httptest.NewRecorder` to construct appropriate arguments for `ServeHTTP` and do it all sequentially with no need for worrying about timeouts, done channels or stopping a server.